### PR TITLE
Dockered Postgres support

### PR DESCRIPTION
**Pull request by kewitz**
Hey, the set-up is pretty straightforward, but I really have trouble with requiring the user to run Postgres locally.

Assumptions:
- Default Postgres instance doesn't requires password.
- Default Postgres instance user is called `postgres`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,14 +14,18 @@ If you see a step below that could be improved (or is outdated), please update t
 ### Prerequisite
 
 1. Make sure you have Node.js version >= 10.
-
-- We recommend using [nvm](https://github.com/creationix/nvm): `nvm use`.
```
Comment (znarf) [inline README.md]: `nvm use` read the `.nvmrc` version, it's great

Comment (kewitz) [reply README.md]: Niiice!

File: scripts/db_restore.sh [hunk 2]
```diff
@@ -48,40 +48,40 @@ if [ -z "$LOCALDBNAME" ]; then usage; fi;
 # where pg_stat_activity.datname = '$LOCALDBNAME'
 # EOF
 
-dropdb --if-exists $LOCALDBNAME;
-createdb $LOCALDBNAME 2> /dev/null
+dropdb -h localhost -U postgres --if-exists $LOCALDBNAME;
```
Comment (znarf) [inline scripts/db_restore.sh]: Why is `-h localhost -U postgres` needed for every command here and afterwards? What would be the alternative?

File: README.md [hunk 3]
```diff
@@ -14,14 +14,17 @@ If you see a step below that could be improved (or is outdated), please update t
...
 2. Make sure you have a PostgreSQL database available
-
-- Check the version: 10.3, 9.6.8, 9.5.12, 9.4.17, 9.3.22 or newer
-- Check that the [PostGIS](https://postgis.net/install/) extension is available
-
+  - Check the version: 10.3, 9.6.8, 9.5.12, 9.4.17, 9.3.22 or newer
+  - Check that the [PostGIS](https://postgis.net/install/) extension is available
+  - Or run it in [Docker](https://www.docker.com/):
+    ```
+    docker run -p 5432:5432 -d --name opencollective-postgres mdillon/postgis:9.6
```
Comment (znarf) [inline README.md]: I would prefer to have that explained in the Postgres.md, maybe with more details. We can add a link to that document here.

File: README.md [hunk 4]
```diff
@@ -14,14 +14,17 @@ If you see a step below that could be improved (or is outdated), please update t
...
-- We recommend using [nvm](https://github.com/creationix/nvm): `nvm use`.
-
+  - We recommend using [nvm](https://github.com/creationix/nvm):
+    ```
+    nvm use
+    ```
 2. Make sure you have a PostgreSQL database available
-
-- Check the version: 10.3, 9.6.8, 9.5.12, 9.4.17, 9.3.22 or newer
-- Check that the [PostGIS](https://postgis.net/install/) extension is available
```
Comment (znarf) [inline README.md]: The formatting should be done with `prettier`. I'll submit a PR to enforce that.

**Comment by znarf** [comment]
I did test and it worked locally with my setup.

File: scripts/db_restore.sh (hunk 2 above)
Comment (kewitz) [reply scripts/db_restore.sh]: Thanks for leading me to a much better solution!

**Comment by kewitz** [comment]
Fixed and rebased.

File: docs/postgres.md [hunk 5]
```diff
@@ -13,6 +13,24 @@ Using brew was not an option:
 - `brew install postgresql postgis` would end up with Postgres 10.x
 - `brew install postgresql@9.x` would end up with Postgres 9.x without possibility to install Postgis
 
+### Using Docker
+
+If you don't want to run a local instance of Postgres in your computer, you can run one in Docker.
+Keep in mind that you still need to have the local client tools like `psql`, `dropdb`, `createuser` still locally available.
```
Comment (znarf) [inline docs/postgres.md]: "still" two times.

Comment (znarf) [inline docs/postgres.md]: When following these steps, how are you making the CLI tools available locally then?

File: docs/postgres.md [hunk 6]
```diff
@@ -13,6 +13,24 @@ Using brew was not an option:
...
+
+```
+docker run -p 5432:5432 -d --name opencollective-postgres mdillon/postgis:9.6
+```
+
+Set the necessary environment variables:
+
+```
+export PGHOST=localhost
+export PGUSER=postgres
```
Comment (znarf) [inline docs/postgres.md]: When does that need to be exported? Everytime before running the project?

**Comment by znarf** [comment]
Will merge in https://github.com/opencollective/opencollective-api/pull/2000

**Pull request closed without merging**